independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/AsyncInputStream.java

```diff
@@ -0,0 +1,201 @@
+package org.jboss.resteasy.reactive.client.handlers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.streams.ReadStream;
+import io.vertx.core.streams.impl.InboundBuffer;
+
+/**
+ * Adapt an InputStream to a ReadStream that can be used with a Pump in Vertx.
+ */
+public class AsyncInputStream implements ReadStream<Buffer>, AutoCloseable {
+    public static final String INPUTSTREAM_IS_CLOSED = "Inputstream is closed";
+    // Based on the inputStream with the real data
+    private final InputStream in;
+    private final Context context;
+    private final InboundBuffer<Buffer> queue;
```

Comment:
using this is _ok_ but it remains an internal construct of vertx that might change.

Reply:
I agree. We could use a standard queue (or a concurrent one)